atomate/vasp/workflows/base/exchange.py

```diff
@@ -0,0 +1,353 @@
+# coding: utf-8
+
+import os
+import numpy as np
+
+from fireworks import Workflow, Firework
+from atomate.vasp.fireworks.core import StaticFW
+from atomate.vasp.powerups import (
+    add_tags,
+    add_modify_incar,
+    add_additional_fields_to_taskdocs,
+    add_wf_metadata,
+    add_common_powerups,
+)
+from atomate.vasp.workflows.base.core import get_wf
+from atomate.vasp.fireworks.exchange import HeisenbergModelFW, VampireCallerFW
+from atomate.vasp.firetasks.exchange_tasks import ExchangeToDB
+from atomate.vasp.config import VASP_CMD, DB_FILE, ADD_WF_METADATA
+from atomate.vasp.workflows.presets.scan import wf_scan_opt
+from atomate.utils.utils import get_logger
+
+from uuid import uuid4
+from copy import deepcopy
+
+logger = get_logger(__name__)
+
+from pymatgen.io.vasp.sets import MPStaticSet
+from pymatgen.analysis.structure_matcher import StructureMatcher, ElementComparator
+from pymatgen.analysis.magnetism.analyzer import (
+    CollinearMagneticStructureAnalyzer,
+    MagneticStructureEnumerator,
+)
+
+__author__ = "Nathan C. Frey"
+__maintainer__ = "Nathan C. Frey"
+__email__ = "ncfrey@lbl.gov"
+__status__ = "Development"
+__date__ = "July 2019"
+__version__ = 0.1
+
+
+class ExchangeWF:
+    def __init__(
+        self,
+        magnetic_structures,
```

Comment:
Can you add a check in the init that all structures have magnetism information and if not raise an error.